bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java

```diff
@@ -174,6 +165,18 @@ protected synchronized void modified(Map<String, Object> properties) {
         }
     }
 
+    private void delayedInitialize(ScheduledExecutorService executor) {
+        // we wait until no further new links or items are announced in order to avoid creation of
+        // items which then must be removed again immediately.
+        final long diff = System.nanoTime() - lastUpdate - TimeUnit.SECONDS.toNanos(2);
+        if (diff < 0) {
```

Comment:
E.g.
* `delayedInitialize` is called fist
* calculate the diff
* The diff result (-1.000.000.000) indicates that there is still 1 second to a 2 second delay
* The method `delayedInitialize` is scheduled again with a delay of 1 second (`-diff`)
* After about 1 second the scheduler will execute the scheduled task `delayedInitialize` again
* calculate the diff
* the diff is now >= 0
* initialize is called

If diff is < 0 again (because `lastUpdate` has been updated), the `delayedInitialize` function is scheduled again.

So the diff should be calculated, the delta is used to schedule another diff calculation if necessary.